src/App.jsx

```diff
@@ -69,6 +111,28 @@ const App = () => {
     });
   };
 
+  useEffect(() => {
+    [cadtRef, climateExplorerRef, climateTokenizationRef].forEach((ref) => {
+      if (ref.current) {
+        handleIframeLoad(ref.current);
+      }
+    });
+  }, [showConnect, theme]);
+
+  useEffect(() => {
+    const fetchTheme = async () => {
+      try {
+        const themeResponse = await fetch("/theme.json");
+        if (themeResponse.ok) {
+          const customTheme = await themeResponse.json();
+          setTheme(customTheme);
```

Comment:
please set the custom theme in a separate state. the theme should not be sent to the child UI's unless it is a custom theme loaded from theme.json

Reply:
have look at line 92, theme wont be sent to the child ui's unless there's a loaded theme from theme.json.

Reply:
my apologies, i misread that. thank you for the clarification